src/victory-util/add-events.js

```diff
@@ -1,9 +1,41 @@
 import React from "react";
-import { defaults, assign, isFunction, partialRight, pick, without } from "lodash";
+import {
+  defaults, assign, keys, isFunction, partialRight, pick, without, isEqual, isEmpty, isPlainObject
+} from "lodash";
 import Events from "./events";
 import VictoryTransition from "../victory-transition/victory-transition";
 
-export default (WrappedComponent) => {
+//  used for checking state changes. Expected components can be passed in via options
+const defaultComponents = [
+  { name: "parent", index: "parent" }, { name: "data" }, { name: "labels" }
+];
+
+const areVictoryPropsEqual = (a, b) => {
```

Comment:
Something like this should have a ton of unit tests behind it. A recursive equality check is something that it's so easy to have incorrect edge cases with.

Perhaps ticket out for "soon" if not now?

/cc @boygirl @chrisbolin